pkg/controllers/startup_policy.go

```diff
@@ -0,0 +1,92 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	jobset "sigs.k8s.io/jobset/api/jobset/v1alpha2"
+)
+
+type startupPolicyAction int64
+
+const (
+	successful startupPolicyAction = 1
+	wait       startupPolicyAction = 2
+	create     startupPolicyAction = 3
+	resume     startupPolicyAction = 4
+)
+
+func matchStartupPolicy(js *jobset.JobSet, rjJobStatus jobset.ReplicatedJobStatus) startupPolicyAction {
+
+	targetedReplicas := getReplicasFromName(js.Spec.ReplicatedJobs, rjJobStatus.Name)
+
+	if rjJobStatus.Ready == targetedReplicas || rjJobStatus.Failed == targetedReplicas || rjJobStatus.Succeeded == targetedReplicas {
+		return successful
+	}
+	if rjJobStatus.Suspended == targetedReplicas {
+		return resume
+	}
+	if rjJobStatus.Ready == 0 && rjJobStatus.Active == 0 && rjJobStatus.Failed == 0 && rjJobStatus.Succeeded == 0 && rjJobStatus.Suspended == 0 {
+		return create
+	}
+	return wait
+}
```

Comment:
This is very confusing and I don't think we need it actually. If we are iterating over the replicatedJobs list in order in both `createJobs` and `resumeJobSetIfNecessary`, then the logic in both functions should look like this:

```
for _, rjob := range js.Spec.ReplicatedJobs {
  if rjob.Replicas == rjJobStatus.Ready + rjJobStatus.Failed + rjJobStatus.Succeeded {
     continue
  }
  unsuspend or create rjob jobs 
  if inOrderStartup {
    break
  }
}
```

Reply:
I applied this idea somewhat. I still think we need to look at previous job to determine if we should start the next job in the sequence.

break out of the loop will say that we were complete so I went with a return to trigger a condition.

Reply:
We worked through this.